### PR TITLE
- added validation for supported twitter accounts

### PR DESCRIPTION
**Pull request by hursevich**

### CODE REVIEW

File: web/src/main/java/com/zebrunner/reporting/web/SocialController.java [hunk 1]
```diff
@@ -18,11 +20,17 @@
...
+    private static final String ZEBRUNNER_NEWS = "zebrunner_news";
+
     private final TwitterService twitterService;
 
     @GetMapping("/twitter/timeline")
     @Override
-    public List<Tweet> getUserTimeline(@RequestParam("userName") String userName, @RequestParam(value = "pageSize", defaultValue = "20") int pageSize) {
+    public List<Tweet> getUserTimeline(@RequestParam(value = "userName", defaultValue = ZEBRUNNER_NEWS) String userName, @RequestParam(value = "pageSize", defaultValue = "20") int pageSize) {
+        // Verify allowed user names
+        if(!Arrays.asList(ZEBRUNNER_NEWS).contains(userName)) {
```
Comment (tsvrko) [inline web/src/main/java/com/zebrunner/reporting/web/SocialController.java:31]: Probably such kind of validation should happen on the service level?

Comment (hursevich) [reply web/src/main/java/com/zebrunner/reporting/web/SocialController.java:31]: Is that common experience to make validation on service layer?

Comment (hursevich) [reply web/src/main/java/com/zebrunner/reporting/web/SocialController.java:31]: I would like to leave validation logic at controller layer.

Comment (nsidorevich) [reply web/src/main/java/com/zebrunner/reporting/web/SocialController.java:31]: Move validation to service please. That way you can make sure that whoever calls that service won't use usernames we don't allow. And yes, it is common approach nowadays :) Typically we try to keep layer responsible only for web-specific things - dto to domain conversion, request params processing, etc

File: web/src/main/java/com/zebrunner/reporting/web/SocialController.java [hunk 2]
```diff
@@ -18,11 +20,17 @@
...
 public class SocialController extends AbstractController implements SocialDocumentedController {
 
+    private static final String ZEBRUNNER_NEWS = "zebrunner_news";
+
     private final TwitterService twitterService;
 
     @GetMapping("/twitter/timeline")
     @Override
-    public List<Tweet> getUserTimeline(@RequestParam("userName") String userName, @RequestParam(value = "pageSize", defaultValue = "20") int pageSize) {
+    public List<Tweet> getUserTimeline(@RequestParam(value = "userName", defaultValue = ZEBRUNNER_NEWS) String userName, @RequestParam(value = "pageSize", defaultValue = "20") int pageSize) {
```
Comment (nsidorevich) [inline web/src/main/java/com/zebrunner/reporting/web/SocialController.java:29]: Let's use `v1/social` as a controller-level mapping and `tweets` as a method level mapping. That way you can extend this controller with other social integrations in future. All new APIs should be versioned ideally. api part will be resolved at a gateway level (we already have this configuration in place)

File: service/src/main/java/com/zebrunner/reporting/service/TwitterService.java [hunk 3]
```diff
@@ -1,20 +1,41 @@
...
-    public List<Tweet> getUserTimeline(String userName, int pageSize) {
+    /**
+     * Read user latest tweets
+     * @param userName - twitter user name
+     * @param pageSize - amount of latest tweets to read
+     * @return user tweets
+     */
+    public List<Tweet> getUserTweets(String userName, int pageSize) {
+        // Verify allowed user names
+        if(!Arrays.asList(ZEBRUNNER_NEWS).contains(userName)) {
```
Comment (tsvrko) [inline service/src/main/java/com/zebrunner/reporting/service/TwitterService.java:32]: Space after if?

**Pull request merged**